### ISSUE TITLE
Implement `Deserializer` for `Pod`

### ISSUE THREAD

**Issue by kmaasrud** [the-alchemists-of-arland/gray-matter-rs#5]
While investigating #3, I realised that we don't really have a need for the polyglot datatype [`Pod`](https://docs.rs/gray_matter/0.2.0/gray_matter/enum.Pod.html). Since we nonetheless coerce it into a [`serde_json::Value`](https://docs.serde.rs/serde_json/value/enum.Value.html#impl-Serialize) upon deserializing, [`Pod`](https://docs.rs/gray_matter/0.2.0/gray_matter/enum.Pod.html) can easily be replaced by [`Value`](https://docs.serde.rs/serde_json/value/enum.Value.html#impl-Serialize) - essentially skipping a conversion step.

This would increase performance (probably not by much, but every nanosecond counts) and drastically reduce code complexity.

Changing this would introduce an API change, so I don't see the rush to do it, but I'm leaving this issue here for whenever we deem it fit to implement.

**Comment by yuchanns** [comment]
Well, when I wrote this crate, I was expecting a scenario where the frontmatter of markdown is parsed out at compile time so that yew can generate static files, which is similar to what vuepress does. So I didn't care about performance, since it's a compile-time overhead.

However, I have noticed that some people use it at runtime, such as #4.

So I general totally agree with you that a little bit of runtime performance optimization is needed.

But as to whether discarding pod types to improve performance, some benchmark comparing this two way shall be done to make it more convincing.

When the pod type was introduced into the crate, I hadn't considered serd_json so I didn't use value instead.

**Comment by yuchanns** [comment]
Also, I think we should do optimization as soon as it is proven necessary. The fewer users, the less cost to make breaking changes. Just follow semantics well.

**Comment by kmaasrud** [comment]
Thinking more about it, perhaps we should keep `Pod` after all! It's not a bad thing to be in control of our own datatype (if we for instance want to extend into supporting TOML's datetimes).

The `deserialize` function is essentially what needs to be changed. Namely, we need to implement `Deserialize` on `Pod` to avoid the need for a conversion into `serde_json::Value`. This wouldn't introduce an API change either :smile:

**Comment by yuchanns** [comment]
Yes, we should think about this a bit more carefully. Currently using serde_json to do serialization and deserialization is just for the sake of meet my need quickly, so I didn't bother with that part, it's a bit complicated after all.
